compiler/rustc_trait_selection/src/solve/mod.rs

```diff
@@ -398,6 +418,103 @@ impl<'a, 'tcx> EvalCtxt<'a, 'tcx> {
             None => self.make_canonical_response(Certainty::AMBIGUOUS),
         }
     }
+
+    #[instrument(level = "debug", skip(self), ret)]
+    fn compute_alias_eq_goal(
+        &mut self,
+        goal: Goal<'tcx, (ty::Term<'tcx>, ty::Term<'tcx>)>,
+    ) -> QueryResult<'tcx> {
+        let tcx = self.tcx();
+
+        let evaluate_normalizes_to = |ecx: &mut EvalCtxt<'_, 'tcx>, alias, rigid| {
+            debug!(
+                "compute_alias_eq_goal::evaluate_normalizes_to(\n alias={:?},\n rigid={:?},\n)",
+                alias, rigid
+            );
+            let r = ecx.infcx.probe(|_| {
+                let (_, certainty) = ecx.evaluate_goal(goal.with(
+                    tcx,
+                    ty::Binder::dummy(ty::ProjectionPredicate {
+                        projection_ty: alias,
+                        term: rigid,
+                    }),
+                ))?;
+                ecx.make_canonical_response(certainty)
+            });
+            debug!("compute_alias_eq_goal::evaluate_normalizes_to(..) -> {:?}", r);
+            r
+        };
+
+        if goal.predicate.0.is_infer() || goal.predicate.1.is_infer() {
+            bug!(
+                "`AliasEq` goal with an infer var on lhs or rhs which should have been instantiated"
+            );
```

Comment:
```suggestion
            bug!("alias eq infer should simply instantiate the inference var");
```

Reply:
I dont like this change it seems far less clear as to what the problem is. It makes it sound like the implementation of `compute_alias_eg_goal` should be instantiating the infer var rather than that the goal should never have been made in the first place.